modules/vaos/app/services/eps/appointment_service.rb

```diff
@@ -32,11 +32,17 @@ def get_appointment(appointment_id:, retrieve_latest_details: false)
     ##
     # Get appointments data from EPS
     #
+    # @param referral_number [String] Optional referral number to filter appointments
     # @return [Array<Hash>] Array of appointment hashes from EPS
     #
-    def get_appointments
+    def get_appointments(referral_number: nil)
+      params = { patientId: patient_id }
+      params[:referralNumber] = referral_number if referral_number.present?
+
```

Comment:
now scopes by referral number if present in method call

<img width="1127" height="918" alt="Screenshot 2025-10-17 at 1 47 40 PM" src="https://github.com/user-attachments/assets/8184a034-549d-4c17-a295-4f252c761a9d" />